framework/tlvf/yaml/tlvf/ieee_1905_1/tlvLinkMetricQuery.yaml

```diff
@@ -2,29 +2,30 @@
 ---
 _namespace: ieee1905_1
 
-tlvLinkMetricQuery:
+# TLVF does not currently support optional fields and this TLV contains one.
+# As a workaround, we define two different TLVs instead of a single tlvLinkMetricQuery with an
+# optional field. Application must then check the length of received message to know if optional 
+# field is present or not and then create an instance of either of these classes.
+
+tlvLinkMetricQueryAllNeighbors:
   _type: class
   _is_tlv_class : True
   type:
     _type: eTlvType
     _value_const: TLV_LINK_METRIC_QUERY
   length: uint16_t
-  neighbor_type: eNeighborType
+  neighbor_type: eLinkMetricNeighborType
+  link_metrics_type: eLinkMetricsType
+
+tlvLinkMetricQuerySpecificNeighbor:
```

Comment:
This is not named correctly. It will also be used for the All Neighbors case for devices that do include a 00:00:00:00:00:00 MAC address.

Reply:
Renamed tlvLinkMetricQuerySpecificNeighbor with its original name tlvLinkMetricQuery because it's suitable for both use cases: all-neighbors and specific-neighbor